lib/src/10_auth/auth_screen.dart

```diff
@@ -128,10 +152,35 @@ class _FauiAuthScreenState extends State<FauiAuthScreen> {
     );
   }
 
-  Widget _buildCreateAccountScreen(BuildContext context) {
-    final TextEditingController emailController =
-        new TextEditingController(text: this._email);
+  Widget _buildTextBox(TextEditingController controller, FocusNode currentNode,
+      FocusNode nextNode, String fieldName, Future<Widget> Function() submit) {
+    handleKey(RawKeyEvent key) {
+      if (key is RawKeyDownEvent) {
+        if (key.physicalKey.debugName == 'Enter')
+          submit();
+        else
+          setState(() => this._error = "");
+      }
+      if ((key.logicalKey.debugName == 'Tab') && (nextNode != null))
+        nextNode.requestFocus();
```

Comment:
Let's improve readability of this handler a little. Too many conditions.

1. You can return if key is not 'down' in the very beginning:
if (key is RawKeyDownEvent)
{
  return;
}

2.  after this set error to empty without any condition

3. Then check for enter and check for tab.

4. And follow this rule: https://dart.dev/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures